test/Microsoft.Extensions.Logging.Test/ConsoleLoggerTest.cs

```diff
@@ -642,6 +642,38 @@ public void CallingBeginScopeOnLogger_ReturnsNonNullableInstance()
             Assert.NotNull(disposable);
         }
 
+#if NET451
+        private static void DomainFunc()
+        {
+            var t = SetUp(filter: null, includeScopes: true);
+            var logger = t.Item1;
+            using (logger.BeginScope("newDomain scope"))
+            {
+                logger.LogInformation("Test");
+            }
+        }
+
+        [Fact]
+        public void ScopeWithChangingAppDomains_DoesNotAccessUnloadedAppDomain()
+        {
+            // Arrange
+            var t = SetUp(filter: null, includeScopes: true);
+            var logger = t.Item1;
+            var sink = t.Item2;
+            var domain = AppDomain.CreateDomain("newDomain");
+
+            // Act
+            domain.DoCallBack(DomainFunc);
+            AppDomain.Unload(domain);
+            var disposable = logger.BeginScope("Scope1");
+            logger.LogInformation("Test");
+
+            // Assert
+            Assert.NotNull(disposable);
+            Assert.Equal(4, sink.Writes.Count);
```

Comment:
I can't recall how these writes look like...how is the count 4 here? (for example, the DomainFunc creates a new logger and new sink, so we are including it's numbers right?)